test/e2e/framework/pod/wait_test.go

```diff
@@ -0,0 +1,296 @@
+			Name: "[Top Level] pod not running",
+			Output: `INFO: Waiting up to 5s for pod "pending-pod" in namespace "default" to be "running"
+INFO: Pod "pending-pod": Phase="", Reason="", readiness=false. Elapsed: <elapsed>
+INFO: Pod "pending-pod": Phase="", Reason="", readiness=false. Elapsed: <elapsed>
+INFO: Pod "pending-pod": Phase="", Reason="", readiness=false. Elapsed: <elapsed>
+INFO: Pod "pending-pod": Phase="", Reason="", readiness=false. Elapsed: <elapsed>
+INFO: Unexpected error:
+    <*pod.timeoutError>: {
+        msg: "timed out while waiting for pod default/pending-pod to be running: last observed as:\n    <*v1.Pod>: {\n        TypeMeta: {Kind: \"\", APIVersion: \"\"},\n        ObjectMeta: {\n            Name: \"pending-pod\",\n            GenerateName: \"\",\n            Namespace: \"default\",\n            SelfLink: \"\",\n            UID: \"\",\n            ResourceVersion: \"\",\n            Generation: 0,\n            CreationTimestamp: {\n                Time: 0001-01-01T00:00:00Z,\n            },\n            DeletionTimestamp: nil,\n            DeletionGracePeriodSeconds: nil,\n            Labels: nil,\n            Annotations: nil,\n            OwnerReferences: nil,\n            Finalizers: nil,\n            ManagedFields: nil,\n        },\n        Spec: {\n            Volumes: nil,\n            InitContainers: nil,\n            Containers: nil,\n            EphemeralContainers: nil,\n            RestartPolicy: \"\",\n            TerminationGracePeriodSeconds: nil,\n            ActiveDeadlineSeconds: nil,\n            DNSPolicy: \"\",\n            NodeSelector: nil,\n            ServiceAccountName: \"\",\n            DeprecatedServiceAccount: \"\",\n            AutomountServiceAccountToken: nil,\n            NodeName: \"\",\n            HostNetwork: false,\n            HostPID: false,\n            HostIPC: false,\n            ShareProcessNamespace: nil,\n            SecurityContext: nil,\n            ImagePullSecrets: nil,\n            Hostname: \"\",\n            Subdomain: \"\",\n            Affinity: nil,\n            SchedulerName: \"\",\n            Tolerations: nil,\n            HostAliases: nil,\n            PriorityClassName: \"\",\n            Priority: nil,\n            DNSConfig: nil,\n            ReadinessGates: nil,\n            RuntimeClassName: nil,\n            EnableServiceLinks: nil,\n            PreemptionPolicy: nil,\n            Overhead: nil,\n            TopologySpreadConstraints: nil,\n            SetHostnameAsFQDN: nil,\n            OS: nil,\n        },\n        Status: {\n            Phase: \"\",\n            Conditions: nil,\n            Message: \"\",\n            Reason: \"\",\n            NominatedNodeName: \"\",\n            HostIP: \"\",\n            PodIP: \"\",\n            PodIPs: nil,\n            StartTime: nil,\n            InitContainerStatuses: nil,\n            ContainerStatuses: nil,\n            QOSClass: \"\",\n            EphemeralContainerStatuses: nil,\n        },\n    }",
```

Comment:
:vomiting_face: 

This direct error output is always going to be a little gross, but one way you could clean this up is to include the pod directly in the TimeoutError struct, and just dump it when error is called. for example:

```go
type timeoutError struct {
	msg string
	lastObservedPod *v1.Pod  // or runtimeObject if we decide to generalize this
}

func (e *timeoutError) Error() string {
	if e.lastObservedPod == nil {
	  return e.msg
	} else {
	  return fmt.Sprintf("%s: last observed as: %s", e.msg, format.Object(e.lastObservedPod, 1))
	}
}

Reply:
Agreed. We could also make the timeout error generic and let it have a slice of objects that get stored as `interface{}`.

Would it make sense to split out the timeout error into its own package? There are other framework packages which implement polling, they may need the same timeout error eventually.

Reply:
My last commit now uses that approach, but without including the extra information in the error message. I am not adding it there because of the discussion around aggregation of failure messages - I am too worried that this won't work anymore.
I think it is a slight improvement because the overall output is shorter.

The timeout error is generic and could be moved as-is into a separate package, but I haven't done that yet.

If I did, should I use `test/e2e/framework/errors` or `test/e2e/framework/e2eerrors`? The latter avoids having to rename the package on each import to avoid clashes with the standard `errors` package, but it would be inconsistent because the other packages don't do that.

The double `ee` is also unfortunate. Perhaps `framework/e2etypes`?